src/profiles.ts

```diff
@@ -0,0 +1,73 @@
+export default class Profiles {
+    private actObjData : Map<string, Array<string>>;
+    private objActData : Map<string, Array<string>>;
+
+    public constructor() {
+        this.actObjData = new Map<string, Array<string>>();
+        this.objActData = new Map<string, Array<string>>();
+    }
+    
+    /*
+        Load the profiles from a JSON string
+    */
+    public loadFromString(s : string) : void {
```

Comment:
Add `public load(data: object) : void` to this is better.

Reply:
Do you mean to add an interface supported for objects in any type like:
```
public load(data: object) : void {
  if (data instanceof String) {
    // Load from String
  } else if (data instanceof OtherType) {
    // ...
  }
}
```